src/com/oltpbenchmark/api/BenchmarkModule.java

```diff
@@ -159,19 +158,25 @@ public final HikariDataSource getDataSource() {
     // IMPLEMENTING CLASS INTERFACE
     // --------------------------------------------------------------------------
 
-    protected abstract List<Worker> makeWorkersImpl();
+    protected List<Worker> makeWorkersImpl() {
+      ArrayList<Worker> workers = new ArrayList<>();
+      try {
+        List<Worker> terminals = createTerminals();
+        workers.addAll(terminals);
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
 
-    /**
-     * Each BenchmarkModule needs to implement this method to load a sample
-     * dataset into the database. The Connection handle will already be
-     * configured for you, and the base class will commit+close it once this
-     * method returns
-     *
-     * @return TODO
-     */
-    protected abstract Loader makeLoaderImpl();
+      return workers;
+    }
+
+    protected Loader makeLoaderImpl() {
+      return new Loader(this);
+    }
 
-    protected abstract Package getProcedurePackageImpl();
+    protected Package getProcedurePackageImpl() {
```

Comment:
dont know why we need this.
Do you?

Reply:
just some other reflection magic that we should get rid of asap :/